File: C++ REF/stack.cpp
#include<iostream>
using namespace std;
//const int size=10;
class stack
{
      public:
            stack();
             int top;
             int size;
             int *elements;
             void push(int);
             int pop();
             //stack(int);
};
void stack::push(int x)
{
     if(top==size-1)
     cout<<"Stack Full...";
     else
     elements[++top]=x;
}
int stack::pop()
{
    if(top==-1)
               {
                cout<<"Stack Empty..";
                return(-1);
                }
                else
                return(elements[top--]);
}
stack::stack()
{           top=-1;
            cout<<"Enter the size of stack: ";
            cin>>size;
            elements=new int[size];
}
/*stack::stack(int x)
{size=x;}*/
int main()
{
    stack s;
    int i=0;
    char c;
    while(i<s.size)
    {
                   cout<<"Enter the element token number:\n ";
                   cin>>i;
                   s.push(i);
                   cout<<"Do you want to take out token number..Press y or n: ";
                   cin>>c;
                   if(c=='y')
                   cout<<s.pop();
                   cout<<"Exit the Program..press y or n: ";
                   cin>>c;
                   if(c=='y')
                   exit(-1);    
    }
    system("pause");
    return(0);
}
                   
                      

File: Java REF/Stack.java
import java.io.*;
import java.util.*;
class Stack1
{
int top=-1,st[]=new int[5];
void push(int el)
{
st[++top]=el;
}
int pop()
{
return(st[top--]);
}
void display()
{
System.out.println("\nStack elements from top to bottom\n");
for(int i=top;i>=0;i--)
System.out.println(st[i]);
}
boolean isFull()
{
return(top==5-1);
}
boolean isEmpty()
{
return(top==-1);
}
}
class Stack
{
public static void main(String a[])
{
Scanner sc=new Scanner(System.in);
Stack1 s=new Stack1();
int el=0,ch=1;
while(ch!=4)
{
System.out.println("\n1.PUSH\n2.POP\n3.DISPLAY\n4.EXIT");
System.out.println("ENTER YOUR CHOICE");
ch=sc.nextInt();
switch(ch)
{
case 1:if(s.isFull())
	System.out.println("\nstack is full");
	else
	{
	System.out.println("Enter element");
	el=sc.nextInt();
	s.push(el);		
	}break;
case 2:if(s.isEmpty())
	System.out.println("\nstack is empty");
	else
	{
	el=s.pop();
	System.out.println("\nDeleted element = "+el);		
	}break;	
case 3:if(s.isEmpty())
	System.out.println("\nstack is empty");
	else
	s.display();
	break;
case 4:break;
default:System.out.println("\nEnter correct choice");
}
}
}
}